fc/fullcontact.go

```diff
@@ -542,6 +529,84 @@ func (fcClient *fullContactClient) PermissionVerify(permissionRequest *Permissio
 	return ch
 }
 
+/*
+FullContact Verify API - VerfiySignals, takes a MultiFieldRequest and returns a channel of type APIResponse.
+
+Request is converted to JSON and sends an Asynchronous request.
+Response will be avaiable in the VerifySignalsResponse field of APIResponse
+*/
+func (fcClient *fullContactClient) VerifySignals(multifieldRequest *MultifieldRequest) chan *APIResponse {
+	// ch := make(chan *APIResponse)
+	// if multifieldRequest == nil {
+	// 	go sendToChannel(ch, nil, "", NewFullContactError("MultiFieldRequest can't be nil"))
+	// 	return ch
+	// }
+
+	// err := multifieldRequest.validate()
+	// if err != nil {
+	// 	go sendToChannel(ch, nil, "", err)
+	// 	return ch
+	// }
+
+	// reqBytes, err := json.Marshal(multifieldRequest)
+	// if err != nil {
+	// 	go sendToChannel(ch, nil, "", err)
+	// 	return ch
+	// }
+	ch, reqBytes, err := doVerfiyAPIPrerequisites(multifieldRequest)
+	if err != nil {
+		go sendToChannel(ch, nil, "", err)
+		return ch
+	}
+	go fcClient.do(verifySignalsUrl, reqBytes, ch)
+	return ch
+}
+
+/*
+FullContact Verify API - VerfiyMatch, takes a MultiFieldRequest and returns a channel of type APIResponse.
+
+Request is converted to JSON and sends an Asynchronous request.
+Response will be avaiable in the VerifyMatchResponse field of APIResponse
+*/
+func (fcClient *fullContactClient) VerifyMatch(multifieldRequest *MultifieldRequest) chan *APIResponse {
+	ch, reqBytes, err := doVerfiyAPIPrerequisites(multifieldRequest)
+	if err != nil {
+		go sendToChannel(ch, nil, "", err)
+		return ch
+	}
+	go fcClient.do(verifyMatchUrl, reqBytes, ch)
+	return ch
+}
+
+/*
+FullContact Verify API - VerfiyActivity, takes a MultiFieldRequest and returns a channel of type APIResponse.
+
+Request is converted to JSON and sends an Asynchronous request.
+Response will be avaiable in the VerifyActivityResponse field of APIResponse
+*/
+func (fcClient *fullContactClient) VerifyActivity(multifieldRequest *MultifieldRequest) chan *APIResponse {
+	ch, reqBytes, err := doVerfiyAPIPrerequisites(multifieldRequest)
+	if err != nil {
+		go sendToChannel(ch, nil, "", err)
+		return ch
+	}
+	go fcClient.do(verifyActivityUrl, reqBytes, ch)
+	return ch
+}
+
+func doVerfiyAPIPrerequisites(multifieldRequest *MultifieldRequest) (chan *APIResponse, []byte, error) {
```

Comment:
We can make this method maybe "validate and marshall multifieldreq" and use it in permission too?

Reply:
Made the change. In addition to it, moved the request call also to the new method